core/src/main/java/smashdudes/ecs/systems/MovementSystem.java

```diff
@@ -23,6 +23,14 @@ public void updateEntity(Entity entity, float dt)
         p.position.x += v.velocity.x * dt;
         p.position.y += v.velocity.y * dt;
 
-        v.velocity.x *= v.deceleration * dt;
+        float decelerationThisFrame = v.deceleration * dt;
+        if (Math.abs(v.velocity.x) > decelerationThisFrame)
```

Comment:
v.velocity.x -= Math.signum(v.velocity.x) * Math.min(Math.abs(v.velocity.x), decelerationThisFrame);
Might be better than here